ImageGalery/Utilities.swift

```diff
@@ -0,0 +1,216 @@
+        var name = named
+        let pathComponents = named.components(separatedBy: "/")
+        if pathComponents.count > 1 {
+            if pathComponents[pathComponents.count-2] == localImagesDirectory {
+                name = pathComponents.last!
+            } else {
+                return nil
+            }
+        }
+        if var url = FileManager.default.urls(for: .applicationSupportDirectory, in: .userDomainMask).first {
+            url = url.appendingPathComponent(localImagesDirectory)
+            do {
+                try FileManager.default.createDirectory(at: url, withIntermediateDirectories: true)
+                url = url.appendingPathComponent(name)
+                if url.pathExtension != "jpg" {
+                    url = url.appendingPathExtension("jpg")
+                }
+                return url
+            } catch let error {
+                print("UIImage.urlToStoreLocallyAsJPEG \(error)")
+            }
+        }
+        return nil
+    }
+    
+    func storeLocallyAsJPEG(named name: String) -> URL? {
+        if let imageData = UIImageJPEGRepresentation(self, 1.0) {
+            if let url = UIImage.urlToStoreLocallyAsJPEG(named: name) {
+                do {
+                    try imageData.write(to: url)
+                    return url
+                } catch let error {
+                    print("UIImage.storeLocallyAsJPEG \(error)")
+                }
+            }
+        }
+        return nil
+    }
+}
+
+extension String {
+    func madeUnique(withRespectTo otherStrings: [String]) -> String {
+        var possiblyUnique = self
+        var uniqueNumber = 1
+        while otherStrings.contains(possiblyUnique) {
+            possiblyUnique = self + " \(uniqueNumber)"
+            uniqueNumber += 1
+        }
+        return possiblyUnique
+    }
+}
+
+extension Array where Element: Equatable {
+    var uniquified: [Element] {
+        var elements = [Element]()
+        forEach { if !elements.contains($0) { elements.append($0) } }
+        return elements
+    }
+}
+
+extension NSAttributedString {
+    func withFontScaled(by factor: CGFloat) -> NSAttributedString {
+        let mutable = NSMutableAttributedString(attributedString: self)
+        mutable.setFont(mutable.font?.scaled(by: factor))
+        return mutable
+    }
+    var font: UIFont? {
+        get { return attribute(.font, at: 0, effectiveRange: nil) as? UIFont }
+    }
+}
+
+extension String {
+    func attributedString(withTextStyle style: UIFontTextStyle, ofSize size: CGFloat) -> NSAttributedString {
+        let font = UIFontMetrics(forTextStyle: .body).scaledFont(for: UIFont.preferredFont(forTextStyle: .body).withSize(size))
+        return NSAttributedString(string: self, attributes: [.font:font])
+    }
+}
+
+extension NSMutableAttributedString {
+    func setFont(_ newValue: UIFont?) {
+        if newValue != nil { addAttributes([.font:newValue!], range: NSMakeRange(0, length)) }
```

Comment:
Put a space after ':'

Reply:
refactor this to 
guard let newSetValue = newValue else { return }
addAttributes([.font: newSetValue], range: NSMakeRange(0, length))
You avoid force unwrapping optional value, which you should always do